### PR TITLE
Rewrite PushingToViews

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Review by KochetovNicolai — COMMENTED** [review]
LGTM

File: src/DataStreams/PushingToViewsBlockOutputStream.cpp [hunk 1]
```diff
@@ -70,428 +178,502 @@ PushingToViewsBlockOutputStream::PushingToViewsBlockOutputStream(
...
 
-    if (view.runtime_stats.thread_status)
-    {
-        /// Change thread context to store individual metrics per view. Once the work in done, go back to the original thread
-        view.runtime_stats.thread_status->resetPerformanceCountersLastUsage();
-        current_thread = view.runtime_stats.thread_status.get();
+        processors.emplace_front(std::move(copying_data));
+        processors.emplace_back(std::move(finalizing_views));
+        result_chain = Chain(std::move(processors));
+        result_chain.setNumThreads(max_parallel_streams);
```
Comment (Algunenano) [inline src/DataStreams/PushingToViewsBlockOutputStream.cpp:339]: I'm not sure if I'm reading this correctly, but this seems to be ignoring the `settings.parallel_view_processing` set above right? It seems that now `views_data->max_threads` is only used to log, but not to control parallelism. Is that correct and intended?

Comment (KochetovNicolai) [reply src/DataStreams/PushingToViewsBlockOutputStream.cpp:339]: I have probably missed it.
Actually, the idea is that views could be processed always in parallel (or not, which depends only on number of executing threads). But probably the proper number of threads is not set.

Comment (azat) [reply src/DataStreams/PushingToViewsBlockOutputStream.cpp:339]: >I'm not sure if I'm reading this correctly, but this seems to be ignoring the settings.parallel_view_processing set above right?

It should be accounted here - https://github.com/ClickHouse/ClickHouse/blob/3663d23e4d80fc5f1fcb8ff1578f49b18319fd26/src/Interpreters/InterpreterInsertQuery.cpp#L410-L412

>But probably the proper number of threads is not set.

#29786

**Pull request merged**